lgc/patch/ShaderMerger.cpp

```diff
@@ -371,6 +372,7 @@ Function *ShaderMerger::generateLsHsEntryPoint(Function *lsEntryPoint, Function
   args.clear();
   attribs.clear();
   attribs.push_back(Attribute::NoRecurse);
+  new FenceInst(*m_context, AtomicOrdering::Release, SyncScope::System, endLsBlock);
```

Comment:
I think we should send the diff dump to backend. Theoretically, s_barrier (see the next statement) is safe enough to make sure the LDS operations are finished.

Reply:
Jay made me aware of the issue, and I think this may not be true. I think what can happen is that ds_store issues in SIMD0 but is stuck in arbitration, e.g. by claused memory instructions from SIMD1. The s_barrier executes, releases a different wave on SIMD1 which executes ds_load, and then the ds_load gets lucky and gets arbitrated before the ds_store from SIMD0.

Reply:
I think we have some codes that call s_barrier without adding explicit fence. According to the LLVM's change, we have to review those points and might apply the changes? The cases are: HS, NGG, Merged shader sync between shader pair, and mesh shader.